src/logic/operate.js

```diff
@@ -0,0 +1,34 @@
+import Big from 'big.js';
+
+export const operate = (numberOne, numeberTwo, operation) => {
+  const first = new Big(numberOne);
+  const second = new Big(numeberTwo);
+  let result;
+
+  if (operation === '-') {
+    result = first.minus(second);
+    return result;
+  }
+
+  if (operation === '÷') {
+    result = first.div(second);
+    return result;
+  }
+
+  if (operation === '+') {
+    result = first.plus(second);
+    return result;
+  }
+
+  if (operation === 'X') {
+    result = first.times(second);
+    return result;
+  }
+
+  if (operation === '=') {
```

Comment:
- [ ] We need to replace this condition with `%` which turns a number to precentage notation.

![](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone4/performOperate.png)